### PR TITLE
Refactor support scripts to functions

### LINKED ISSUE #678: Refactor Support scripts to functions
# 💡 Summary #

ScubaGear is currently deployed as a zipped package with manual instructions on installation and usage.  To migrate to delivery/deployment via a package repository all needed support scripts should be refactored into functions (inside a module).  

## Motivation and context ##

This would be useful because support publishing to public repository #645 

## Implementation notes ##

Reference #94, #645, #677 for context and additional information 

- Create new module for ScubaGear support 
- Include all needed, refactored support functions (e.g., OPA.ps1)

## Acceptance criteria ##

How do we know when this work is done?

- [x] Current support script, OPA.ps1, executes as a function after installing ScubaGear from a repository
- [x] Current support script, utils/ScubaGearSupport.ps1, executes as a function after installing ScubaGear from a repository
- [x] Current support script, utils/RunScuba.ps1, executes as a function after installing ScubaGear from a repository

### LINKED ISSUE #772: Remove Obsolete helper scripts
# 🐛 Summary #

When running .\RunSCUBA.ps1 getting the following error....

PS C:\tmp\ScubaGear-1.0.0\utils> .\RunSCuBA.ps1
Invoke-SCuBA : Cannot validate argument on parameter 'ProductNames'. The argument "onedrive" does not belong to the set 
"teams,exo,defender,aad,powerplatform,sharepoint,*" specified by the ValidateSet attribute. Supply an argument that is in the set and then try the command again.
At C:\tmp\ScubaGear-1.0.0\utils\RunSCuBA.ps1:33 char:18
+     Invoke-SCuBA @SCuBAParams
+                  ~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Invoke-SCuBA], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,Invoke-SCuBA
 

Add any screenshots of the problem here.

### PR DESCRIPTION
**Pull request by crutchfield**
#  #
  

## 🗣 Description ##

Several changes were made to 'support' scripts for Scuba to include SetUp.ps1, OPA.ps1, et.al.
These changes are in support of packaging all functionality within a single module to be deployed into the public package repository, PowerShell Gallery. 

## 💭 Motivation and context ##

To keep usability easy, we do not want the user to have to maneuver to the installed module directory to find and execute scripts.  This move the scripts into the module as cmdlets.  
Closes #772 
Closes #678  

## 🧪 Testing ##

-Ran unit test
-Ran Invoke-SCuBA
-Ran Install-OPA
-Ran Initialize-SCuBA 
-Manually dispatched workflows: run_runctional_product_tests, run_publish_package, run_smoke_test, run_release

## ✅ Pre-approval checklist ##

- [X] This PR has an informative and human-readable title.
- [X] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [X] Changes are limited to a single goal - *eschew scope creep!*
- [X] Changes are sized such that they do not touch excessive number of files.
- [X] *All* future TODOs are captured in issues, which are referenced in code comments.
- [X] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [X] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [X] All relevant type-of-change labels added.
- [X] All relevant project fields are set.
- [X] All relevant repo and/or project documentation updated to reflect these changes.
- [X] Unit tests added/updated to cover PowerShell and Rego changes.
- [X] Functional tests added/updated to cover PowerShell and Rego changes.
- [X] All relevant functional tests passed.
- [X] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [ ] PR passed smoke test check.
- [ ] Feature branch has been rebased against changes from parent branch, as needed

  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.
- [ ] Resolved all merge conflicts on branch
- [ ] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

File: .github/workflows/run_functional_product_tests.yaml [hunk 1]
```diff
@@ -41,18 +41,23 @@ jobs:
...
-        run: Install-Module -Name Selenium
-      - name: Update Selenium
-        run: ./Testing/Functional/SmokeTest/UpdateSelenium.ps1
+
+      - name: Install Selenium & Update Wed Drivers
+        run: |
+          Install-Module -Name Selenium -Scope CurrentUser -Force
+          Import-Module -Name Selenium
+          ./Testing/Functional/SmokeTest/UpdateSelenium.ps1
+          Remove-Module -Name Selenium
```
Comment (james-garriss) [inline .github/workflows/run_functional_product_tests.yaml:50]: As runners are ephemeral, removing modules are unnecessary and waste time.

File: .github/workflows/run_functional_product_tests.yaml [hunk 2]
```diff
@@ -41,18 +41,23 @@ jobs:
...
       - name: Test Products
         # The testing params are stored as a string in a CSV-like format.
         # Pipes separate products, and commas separate key/value pairs.
         # key1=value1,key2=value2|key3=value3,key4=value4...
         # This job parses the keys and values and converts them to a hashtable.
         run: |
+          Import-Module -Name .\PowerShell\ScubaGear\ScubaGear.psd1
+          Install-OPA
+          # Workaround for Selenium. Loading psm1 instead of psd1
+          Import-Module -Name (Get-Module -Name Selenium -ListAvailable).Path -Force
```
Comment (james-garriss) [inline .github/workflows/run_functional_product_tests.yaml:60]: These should be in a previous setup step, not the step that is parsing the params and calling the product tests.

**Review by james-garriss — COMMENTED** [review]
A few small things to clean up, but overall looks great.  Thanks for editing all of these.

File: .github/workflows/run_functional_product_tests.yaml (hunk 2 above)
Comment (crutchfield) [reply .github/workflows/run_functional_product_tests.yaml:60]: Moved to previous, setup, step

**Comment by crutchfield** [comment]
> @crutchfield No new comments but the smoke test is failing on the branch - maybe because of all the conflicts that need to be resolved (?).

Appears to rebase resolved the Smoke test issue (pinning modules for sharepoint/ms graph)

**Review by james-garriss — APPROVED** [review]

**Review by nanda-katikaneni — APPROVED** [review]

**Comment by crutchfield** [comment]
@nanda-katikaneni Ready to merge to main

**Pull request merged**